File: server/model/Product.js
'use strict';

let bookshelf = require('./bookshelf');
let BaseModel = require('./BaseModel');

require('./Category');

let Product = BaseModel.extend({
    tableName: 'product',
    attrs: {
    	id: {
    		field: 'id'
    	},
    	name: {
    		field: 'name'
    	},
    	browseCount: {
    		field: 'browse_count'
    	},
    	buyCount: {
    		field: 'buy_count'
    	},
    	totalSale: {
    		field: 'total_sale'
    	},
        price: {
            field: 'price'
        },
        originalPrice: {
            field: 'original_price'
        },
        status: {
            field: 'status'
        },
        remark: {
            field: 'remark'
        },
        detail: {
            field: 'detail'
        }
    },
    categories: function() {
        return this.belongsToMany('Category');
    }
}, {

});

Product.UP_SHELF   = 1;
Product.DOWN_SHELF = 2;

module.exports = bookshelf.model('Product', Product);

File: server/model/product.go
package model

import "time"

// Product 商品
type Product struct {
    ID             uint       `gorm:"primary_key" json:"id"`
	CreatedAt      time.Time  `json:"createdAt"`
	UpdatedAt      time.Time  `json:"updatedAt"`
	DeletedAt      *time.Time `sql:"index" json:"deletedAt"`
    Name           string     `json:"name"`
    BrowseCount    int        `json:"browseCount"`
    BuyCount       int        `json:"buyCount"`
    TotalSale      float64    `json:"totalSale"`
    Price          float64    `json:"price"`      
    OriginalPrice  float64    `json:"originalPrice"`
    Status         int        `json:"status"`
    Remark         string     `json:"remark"`
    Detail         string     `json:"detail"`
}